.ci/install_kata_kernel.sh

```diff
@@ -132,8 +137,11 @@ main() {
 			build_and_install_kernel
 		fi
 	else
-		build_and_install_kernel
-
+		if [ "$cached_kernel_version" == "$current_kernel_version" ] && [ "$kernel_arch" == "x86_64" ]; then
```

Comment:
I think these lines are not needed as they run the same functions as above. Also I think we can remove the `if` on L128 as you are already taking into consideration the `experimental_kernel` flag inside the `build_and_install_kernel` and `install_cached_kernel` functions.